From e572e277a812ffbf5e40115011669626b046395e Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 29 Oct 2024 11:34:00 +0100
Subject: [PATCH 1342/1695] dts: stm32: fix GIC interrupt ligns in
 stm32mp211.dtsi

Correct GIC interrupt lines for many peripherals as they were wrongly
copied from the stm32mp251.dtsi file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e18e7e6660ca7c220ae3487b4f8d2bd6214269c9 dts: stm32: fix GIC interrupt ligns in stm32mp211.dtsi]
Change-Id: Ide0f8c6d0030a34f8d97ca87e75dd8095600a413
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/414862
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Cheick TRAORE <cheick.traore@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi  | 20 ++++++++++----------
 core/arch/arm/dts/stm32mp21xc.dtsi |  6 +++---
 2 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index a09dc3942..665eba4bd 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -2085,7 +2085,7 @@
 			usart2: serial@400e0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x400e0000 0x400>;
-				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_USART2_ID>;
 				status = "disabled";
@@ -2094,7 +2094,7 @@
 			usart3: serial@400f0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x400f0000 0x400>;
-				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_USART3_ID>;
 				status = "disabled";
@@ -2103,7 +2103,7 @@
 			uart4: serial@40100000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40100000 0x400>;
-				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_UART4_ID>;
 				status = "disabled";
@@ -2112,7 +2112,7 @@
 			uart5: serial@40110000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40110000 0x400>;
-				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_UART5_ID>;
 				status = "disabled";
@@ -2145,7 +2145,7 @@
 			usart6: serial@40220000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40220000 0x400>;
-				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_USART6_ID>;
 				status = "disabled";
@@ -2154,7 +2154,7 @@
 			usart1: serial@40330000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40330000 0x400>;
-				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_USART1_ID>;
 				status = "disabled";
@@ -2163,7 +2163,7 @@
 			uart7: serial@40370000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x40370000 0x400>;
-				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_UART7_ID>;
 				status = "disabled";
@@ -2240,7 +2240,7 @@
 			hash2: hash@42010000  {
 				compatible = "st,stm32mp13-hash";
 				reg = <0x42010000  0x400>;
-				interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 178 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_HASH2>;
 				resets = <&rcc HASH2_R>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_HASH2_ID>;
@@ -2392,7 +2392,7 @@
 		iac: iac@42090000 {
 			compatible = "st,stm32mp25-iac";
 			reg = <0x42090000 0x400>;
-			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		risaf1: risaf@420a0000 {
@@ -2504,7 +2504,7 @@
 		serc: serc@44080000 {
 			compatible = "st,stm32mp25-serc";
 			reg = <0x44080000 0x1000>;
-			interrupts = <GIC_SPI 240 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc CK_BUS_SERC>;
 		};
 
diff --git a/core/arch/arm/dts/stm32mp21xc.dtsi b/core/arch/arm/dts/stm32mp21xc.dtsi
index 38068a46e..17bbddf13 100644
--- a/core/arch/arm/dts/stm32mp21xc.dtsi
+++ b/core/arch/arm/dts/stm32mp21xc.dtsi
@@ -8,7 +8,7 @@
 	cryp1: cryp@42030000 {
 		compatible = "st,stm32mp1-cryp";
 		reg = <0x42030000 0x400>;
-		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
+		interrupts = <GIC_SPI 131 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_CRYP1>;
 		resets = <&rcc CRYP1_R>;
 		status = "disabled";
@@ -17,7 +17,7 @@
 	hash1: hash@42030400 {
 		compatible = "st,stm32mp13-hash";
 		reg = <0x42030400  0x400>;
-		interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>;
+		interrupts = <GIC_SPI 132 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_HASH1>;
 		resets = <&rcc HASH1_R>;
 		status = "disabled";
@@ -45,7 +45,7 @@
 	pka: pka@42032000 {
 		compatible = "st,stm32mp13-pka64";
 		reg = <0x42032000 0x2000>;
-		interrupts = <GIC_SPI 146 IRQ_TYPE_LEVEL_HIGH>;
+		interrupts = <GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CK_BUS_PKA>, <&rcc CK_BUS_RNG1>;
 		clock-names = "bus", "rng";
 		resets = <&rcc PKA_R>;
-- 
2.47.2

